### PR TITLE
[Feature]  Add to_cugraph and from_cugraph to DGL

### PR DESCRIPTION
**Pull request by VibhuJawa**
## Description

This PR adds  `to_cugraph` and `from_cugraph` to DGL 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

## Changes
- [x] `to_cugraph` 
- [x] Test for `to_cugraph` 
- [x]  `from_cugraph`
- [x] Test for `from_cugraph` 

## Installation instructions of cugraph+Pytorch+DGL

```python
mamba create -n dgl_and_cugraph -c dglteam -c rapidsai-nightly -c nvidia -c pytorch -c conda-forge cugraph=22.08 pytorch torchvision torchaudio cudatoolkit=11.3 dgl-cuda11.3 tqdm
```

### CODE REVIEW

**Comment by VibhuJawa** [comment]
CC : @jermainewang

**Review by jermainewang — APPROVED** [review]
LGTM overall. Some minor suggestions. Thanks for the contribution. Could you also update the API doc?
* For dgl namespace, https://github.com/dmlc/dgl/blob/master/docs/source/api/python/dgl.rst
* For DGLGraph members, https://github.com/dmlc/dgl/blob/master/docs/source/api/python/dgl.DGLGraph.rst

**Comment by jermainewang** [comment]
@Rhett-Ying Could you help install cugraph to our CI so the unit tests can be properly enabled?

File: python/dgl/convert.py [hunk 1]
```diff
@@ -1620,6 +1622,110 @@ def to_networkx(g, node_attrs=None, edge_attrs=None):
...
+        cugraph_graph = cugraph_graph.to_directed()
+
+    edges = cugraph_graph.edges()
+    src_t = F.zerocopy_from_dlpack(edges['src'].to_dlpack())
+    dst_t = F.zerocopy_from_dlpack(edges['dst'].to_dlpack())
+    g = graph((src_t,dst_t))
+
+    return g
+
+DGLHeteroGraph.from_cugraph = from_cugraph
```
[the remark this answers is not in the document]
Comment (VibhuJawa) [reply python/dgl/convert.py]: Resolved with https://github.com/dmlc/dgl/pull/4166/commits/87d9ce26cabc1303c767feb7ae4fcacfc1f66eb7

**Comment by VibhuJawa** [comment]
> LGTM overall. Some minor suggestions. Thanks for the contribution. Could you also update the API doc?
> 
> * For dgl namespace, https://github.com/dmlc/dgl/blob/master/docs/source/api/python/dgl.rst
> * For DGLGraph members, https://github.com/dmlc/dgl/blob/master/docs/source/api/python/dgl.DGLGraph.rst

Thanks for the review , have added the API docs (see [commit](https://github.com/dmlc/dgl/pull/4166/commits/87d9ce26cabc1303c767feb7ae4fcacfc1f66eb7)) and addressed all the reviews. Please let me know if I need to do anything else.

**Comment by Rhett-Ying** [comment]
@VibhuJawa  Hi, is it possible to install cugraph via `conda install -n pytorch-ci -c nvidia -c rapidsai -c numba -c conda-forge cugraph cudatoolkit=11.4` on docker container based on `nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04` ? I tried but failed with below error: 

```
PackagesNotFoundError: The following packages are not available from current channels:

  - cudatoolkit=11.4                                                                                          
  - __glibc[version='>=2.17,<3.0.a0']
```

**Comment by VibhuJawa** [comment]
>  @Rhett-Ying , Cordinating with the team internally to find the right container for you to run this on. I am afraid it wont work on the `nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04` container.

**Comment by chang-l** [comment]
@Rhett-Ying @VibhuJawa  nvidia/pytorch container (https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch) should already have cugraph installed. Just pull the latest release and it should work.

**Comment by jermainewang** [comment]
#4183

**Comment by Rhett-Ying** [comment]
@chang-l  Tried with `22.04` and `cugraph` is imported succesfully.

**Comment by Rhett-Ying** [comment]
@VibhuJawa https://github.com/dmlc/dgl/pull/4171 has been merged. In order to test cugraph related cases, pls put them into `//tests/cugraph/test_basics.py` and rebase on master branch.

File: tests/compute/test_basics.py [hunk 2]
```diff
@@ -290,6 +291,66 @@ def _check_nx_feature(nxg, nf, ef):
     assert F.allclose(g.edata['h'], F.tensor([[1., 2.], [1., 2.],
                                               [2., 3.], [2., 3.]]))
 
+def test_to_cugraph_conversion():
```
Comment (Rhett-Ying) [inline tests/compute/test_basics.py:294]: pls move these tests into `//tests/cugraph/test_basics.py`.

Comment (VibhuJawa) [reply tests/compute/test_basics.py]: Done . Thanks for adding cugraph to the CI.

**Comment by VibhuJawa** [comment]
@jermainewang , I think this is good to merge now.

**Pull request merged**